src/main/java/com/facebook/ads/sdk/Campaign.java

```diff
@@ -1653,6 +1627,51 @@ public APIRequestGetInsights requestField (String field, boolean value) {
 
   }
 
+  public static class APIRequestGetInsights extends APIRequestGetInsightsBase<AdsInsights> {
+    public APIRequestGetInsights(String nodeId, APIContext context) {
+      super(nodeId, context, "GET");
+    }
+
+    @Override
+    public APINodeList<AdsInsights> parseResponse(String response) throws APIException {
+      return AdsInsights.parseResponse(response, getContext(), this);
+    }
+
+    @Override
+    public APINodeList<AdsInsights> execute() throws APIException {
+      return execute(new HashMap<String, Object>());
+    }
+
+    @Override
+    public APINodeList<AdsInsights> execute(Map<String, Object> extraParams) throws APIException {
+      lastResponse = parseResponse(callInternal(extraParams));
+      return lastResponse;
+    }
+  }
+
+  public static class APIRequestGetInsightsAsync extends APIRequestGetInsightsBase<AdReportRun> {
+    public APIRequestGetInsightsAsync(String nodeId, APIContext context) {
+      super(nodeId, context, "POST");
+    }
+
+    @Override
+    public APINodeList<AdReportRun> parseResponse(String response) throws APIException {
+      return AdReportRun.parseResponse(response, getContext(), this);
```

Comment:
@wmoussa This should do the trick. As for adding to examples - I'll definitely do it, but a bit later.